Client/Ecosia/UI/Theme/EcosiaThemeManager.swift

```diff
@@ -187,10 +187,9 @@ public final class EcosiaThemeManager: ThemeManager, Notifiable {
 
 extension EcosiaThemeManager {
     
-    func updateLegacyThemeIfNeeded() {
+    func updateLegacyThemeIfSystemThemeON() {
         if LegacyThemeManager.instance.systemThemeIsOn {
-            let userInterfaceStyle = UIScreen.main.traitCollection.userInterfaceStyle
-            LegacyThemeManager.instance.current = userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
+            LegacyThemeManager.updateBasedOnCurrentSystemThemeType()
```

Comment:
Actually doing this is what bring back the [fatal error from before](https://github.com/ecosia/ios-browser/pull/771#discussion_r1742223716) 🙈 Maybe my comment above is a way to fix it without keeping the code duplication?

Reply:
Certainly need to clean derived data every time before testing it then 🤦‍♂️. Glad you made a run beforehand. Let me have a look 👀